python/graphstorm/gconstruct/transform.py

```diff
@@ -665,6 +666,129 @@ def call(self, feats):
 
         return {self.feat_name: feats}
 
+class NumericalStandardTransform(TwoPhaseFeatTransform):
+    r""" Numerical value with standard normalization.
+
+        $val_i = \frac{val_i}{\sum_{i=0}^{N-1} val_i}$
+
+    Parameters
+    ----------
+    col_name : str
+        The name of the column that contains the feature.
+    feat_name : str
+        The feature name used in the constructed graph.
+    sum: np.array or list of floats
+        The summation of all the values in the feature col
+        from the last transformation.
+        If it is None, we will compute it on the fly.
+        Default: None
+    out_dtype:
+        The dtype of the transformed feature.
+        Default: None, we will not do data type casting.
+        Default: None
+    transform_conf : dict
+        The configuration for the feature transformation.
+        Default: None
+    """
+    def __init__(self,
+                 col_name : str,
+                 feat_name : str,
+                 summation : Optional[Any] = None,
+                 out_dtype: Optional[str] = None,
+                 transform_conf: Optional[dict] = None):
+        self._summation = np.array(summation, dtype=np.float32) if summation is not None else None
+        self._conf = transform_conf
+
+        out_dtype = np.float32 if out_dtype is None else out_dtype
+        super(NumericalStandardTransform, self).__init__(col_name, feat_name, out_dtype)
+
+    def pre_process(self, feats) -> Dict[str, np.array]:
+        assert isinstance(feats, (np.ndarray, ExtMemArrayWrapper)), \
+            f"Feature {self.feat_name} of NumericalMinMaxTransform " \
+            "must be numpy array or ExtMemArray"
+
+        # If sum has already been set.
+        # Skip the pre-process step
+        if self._summation is not None:
+            return {self.feat_name: None}
+
+        if isinstance(feats, ExtMemArrayWrapper):
+            # TODO(xiangsx): This is not memory efficient.
+            # It will load all data into main memory.
+            feats = feats.to_numpy()
+
+        if feats.dtype not in [np.float64, np.float32, np.float16, np.int64, \
+                              np.int32, np.int16, np.int8]:
```

Comment:
I would perfer to change ``np.float64, np.float32, np.float16, np.int64, np.int32, np.int16, np.int8`` into a constant in another PR.

The ``\`` is necessary otherwise, the length of the code will exceed 100.